src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -0,0 +1,129 @@
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::InferenceOnlyTransformerImpl<
+            typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+            std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    ForecastingPivotTransformer(void) = default;
+    ForecastingPivotTransformer(Archive &ar);
+
+    ~ForecastingPivotTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ForecastingPivotTransformer);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using NullableType                      = typename Microsoft::Featurizer::Traits<InputT>::nullable_type;
+    using MatrixType                        = Microsoft::Featurizer::RowMajMatrix<NullableType>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        if (input.empty()) {
+            return;
+        }
+        
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type> ret;
+        Eigen::Index const col_size(input[0].cols());
+
+        for (Eigen::Index col = 0; col < col_size; ++col) {
+            bool has_nan(false);
+
+            // if any nans is found in the current column across all matrixes, this column is omitted
+            for (MatrixType const & matrix : input) {
+                // all matrixes must have the same column number
+                assert(matrix.cols() == col_size);
+
+                for (Eigen::Index row = 0; row < matrix.rows(); ++row) {
+                    if (Traits<NullableType>::IsNull(matrix(row, col))) {
+                        has_nan = true;
+                        break;
+                    }
+
+                    ret.emplace_back(matrix(row, col));
+                }
+
+                if (has_nan) {
+                    break;
+                }
+            }
+
+            if (!has_nan) {
+                callback(ret);
```

Comment:
>ret [](start = 25, length = 3)

`std::move(ret)`

Reply:
@angryjinyan If you do the std::move here make sure you initialize your vector in the for loop and dont call clear at the end.

Reply:
Changed return vector scope to be within the for loop and use `std::move()`. Thank you for your advice @michaelgsharp